test/mpi/configure.ac

```diff
@@ -1269,13 +1270,24 @@ if test "$f90dir" = "f90" ; then
             end
         ])
     ],[
-        AC_MSG_RESULT(yes)
+        has_f90_choice_buffer_routines=yes
     ],[
-        AC_MSG_RESULT(no)
+        has_f90_choice_buffer_routines=no
         # patch f90/profile tests
         find "$srcdir/f90/profile" -name "*.f90" -exec sed -i "s/use mpi, skip_.*/use mpi/" {} \;
     ])
     AC_LANG_POP([Fortran])
+    AC_MSG_RESULT([$has_f90_choice_buffer_routines])
+    for a in `find "$srcdir/f90/profile" -name "*.f90.in"` ; do
+        b=`basename $a .f90.in`
+        b="f90/profile/$b.f90"
+        echo "substituting $b"
+        if test "$has_f90_choice_buffer_routines" = "yes" ; then
+            cp $a $b
+        else
+            sed "s/use mpi, skip_.*/use mpi/" $a > $b
+        fi
+    done
```

Comment:
maybe we just disable this test when choice buffer routines aren't available? i'm not sure its worth all this extra logic. plus the f77 profile test essentially covers the case when no choice buffer routines are available, if i understand correctly.

Reply:
Only configure will know whether we have the routines, so some logic needs to take place in configure other than moving the entire `f77tof90` into configure.

Reply:
I was thinking something like comment out the profile directory in `test/mpi/f90/testlist` with a @f90profile@ placeholder.

Reply:
I see. I can do that.

Reply:
We have `test/mpi/f90/testlist.in` already, so might not be too bad.

Reply:
Done. It is much simpler.